src/java.base/share/classes/sun/launcher/LauncherHelper.java

```diff
@@ -590,9 +590,9 @@ static void initOutput(PrintStream ps) {
         ostream = ps;
     }
 
-    static String getMainClassFromJar(String jarname) {
+    private static String getMainClassFromJar(String jarname, JarFile jarFile) {
```

Comment:
Can you check if the "jarname" parameter can be dropped, the error handling in this method should be able to use jarFile.getName().

Reply:
`String jarname` is filled by the caller with the value of `String what`, and therefore contains the entire path to the executable JAR file. Using only `jarFile.getName()` will prevent any parent directories from being noted in error messages; which might affect some tests

Reply:
> `String jarname` is filled by the caller with the value of `String what`, and therefore contains the entire path to the executable JAR file. Using only `jarFile.getName()` will prevent any parent directories from being noted in error messages; which might affect some tests

It returns the path name so I think it does what we want. Do you observe a behavior difference?

Reply:
No, I don't.

Also double-checked via `JarFile/ZipFile::getName`'s implementation returning the entire path to the JAR file.

Re-running tier1-3 tests just in case...